pkg/loader/fileloader_test.go

```diff
@@ -28,11 +28,11 @@ func TestFileLoaderHappyPath(t *testing.T) {
 	location := "foo"
 	content := []byte("bar")
 	fakefs.WriteFile(location, content)
-	l, err := NewFileLoader(fakefs, location)
```

Comment:
We might want to only test the external interface, and drop these tests.

Reply:
Can't agree on this.
Ideally we should have high TC on everything.

Reply:
I think we should only test the implementation details if we can't test them through the public interface. But, yeah, nvm

Reply:
I'll rephrase. We hope to have a stable external interface, but we're probably going to change the implementation many times. A stronger set of external test would allow us to improve the implementation more quickly, because we wouldn't have to change any tests.

it means that:
- If the tests are broken, it's because the implementation is broken, not because it's different.
- the tests are very focused on the exact behavior provided by the interface. If we have many users of this interface across the project, they need to know precisely how it works.

These conditions means that we have more confidence when changing the code, and also that we don't have to maintain a bunch of tests that keeps breaking.

Nuance to that of course: some things are hard to test because you need to be able to mock some specific knobs in your implementation. I'm not opposed to that, but they should be the exception rather than the rules. I don't think the tests we have here are the exception.